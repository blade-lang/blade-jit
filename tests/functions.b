var a = 'outer'

def test() {
  var a = 'inner'
  echo 'It works! ${a}'
}

echo a

test()

def test2(name, age, ...args) {
  echo name
  echo age
  echo args
}

test2('Richard', 20, 'James')

def sin(n) {
  if n {
    var t = n, sine = t

    iter var a = 1; a < 24; a++ {
      var mult = -n * n / ((2 * a + 1) * (2 * a))
      t *= mult
      sine += t
    }

    return sine
  }
  return nil
}

echo sin()
echo 'Sin 10 = ${sin(10)}'
