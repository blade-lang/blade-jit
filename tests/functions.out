outer
It works! inner
Richard
20
[James]
nil
Sin 10 = -0.5440211108893656